gusto/time_discretisation.py

```diff
@@ -323,13 +323,37 @@ def setup(self, equation, apply_bcs=True, *active_labels):
 
         super().setup(equation, apply_bcs, *active_labels)
 
+        # Get continuity form transport term
+        for t in self.residual:
+            if(t.get(transport) == TransportEquationType.conservative):
+                    # Split continuity form term
+                    test = t.form.arguments()[0]
+                    subj = t.get(subject)
+                    prognostic_field_name = t.get(prognostic)
+                    idx = self.equation.field_names.index(prognostic_field_name)
+                    transported_field = split(subj)[idx]
+                    # u_idx = self.equation.field_names.index('u')
+                    # uadv = split(self.equation.X)[u_idx]
+                    # breakpoint()
+                    uadv = t.get(transporting_velocity)
+                    breakpoint()
+                    new_transport_term = prognostic(subject(advection_form(test, transported_field, uadv) + test*transported_field*div(uadv)*dx, subj, prognostic_field_name))
+                    # Add onto residual and drop old term
+                    self.residual = self.residual.label_map(
+                        lambda t: t.get(transport) == TransportEquationType.conservative,
+                        map_if_true=drop)
+                    self.residual += new_transport_term.form
```

Comment:
you should just add the labelled form to the residual, not just the form part, i.e. `self.residual += new_transport_term`